integration/wrappers-regression/google/protobuf/timestamp.ts

```diff
@@ -200,16 +199,15 @@ type KeysOfUnion<T> = T extends T ? keyof T : never;
 export type Exact<P, I extends P> = P extends Builtin ? P
   : P & { [K in keyof P]: Exact<P[K], I[K]> } & { [K in Exclude<keyof I, KeysOfUnion<P>>]: never };
 
-function longToNumber(long: Long): number {
-  if (long.gt(globalThis.Number.MAX_SAFE_INTEGER)) {
+function longToNumber(int64: { toString(): string }): number {
+  const num = globalThis.Number(int64.toString());
+  if (num > globalThis.Number.MAX_SAFE_INTEGER) {
     throw new globalThis.Error("Value is larger than Number.MAX_SAFE_INTEGER");
   }
-  return long.toNumber();
-}
-
-if (_m0.util.Long !== Long) {
-  _m0.util.Long = Long as any;
-  _m0.configure();
```

Comment:
@timostamm might be a moot point in the long run, but I think this "hook up Long to protobufjs and call configure" is required, and had been using a ts-poet-ism to get like auto-inserted into the output on the conditional use of `Long` in the output...

So the otherwise-lgtm change of using the `{ toString(): string }` meant ts-poet doesn't know to implicitly add this anymore (which is admittedly kind of cute/too-cute on how its gets added).

Am I missing something, where this isn't actually required? The intent was just to have it "just work" for users, and not need to do this themselves. If we should keep it, we can probably figure out another way to get it conditionally into the output...

Reply:
Let's add it back 👍  I'll take a look tomorrow.

Reply:
Uh, I remember what's going on with protobuf.js and Long.js, and I wish I didn't.

`protobufjs` automagically loads `long` from node_modules, and uses it in `Reader`. If the magic fails (easily happens with bundlers), it falls back to an alternative and returns Number instead of Long.

We could hook the ConditionalOutput to `Reader.int64` instead. Then we're free to refactor longToNumber.